lib/pharos/cluster_manager.rb

```diff
@@ -45,51 +48,34 @@ def validate
       apply_phase(Phases::ValidateHostname, config.hosts, ssh: false, parallel: false)
     end
 
-    # @return [Array<Pharos::Configuration::Host>]
-    def sorted_master_hosts
-      config.master_hosts.sort_by(&:master_sort_score)
-    end
-
-    # @return [Array<Pharos::Configuration::Host>]
-    def sorted_etcd_hosts
-      config.etcd_hosts.sort_by(&:etcd_sort_score)
-    end
-
     def apply_phases
-      # we need to use sorted masters because phases expects that first one has
-      # ca etc config files
-      master_hosts = sorted_master_hosts
-
-      apply_phase(Phases::MigrateMaster, master_hosts, ssh: true, parallel: true)
+      apply_phase(Phases::MigrateMaster, config.master_hosts, ssh: true, parallel: true)
       apply_phase(Phases::ConfigureHost, config.hosts, ssh: true, parallel: true)
 
-      unless @config.etcd&.hosts
-        # we need to use sorted etcd hosts because phases expects that first one has
-        # ca etc config files
-        etcd_hosts = sorted_etcd_hosts
-        apply_phase(Phases::ConfigureCfssl, etcd_hosts, ssh: true, parallel: true)
-        apply_phase(Phases::ConfigureEtcdCa, [etcd_hosts.first], ssh: true, parallel: false)
-        apply_phase(Phases::ConfigureEtcdChanges, [etcd_hosts.first], ssh: true, parallel: false)
-        apply_phase(Phases::ConfigureEtcd, etcd_hosts, ssh: true, parallel: true)
+      if config.etcd_hosts?
+        apply_phase(Phases::ConfigureCfssl, config.etcd_hosts, ssh: true, parallel: true)
+        apply_phase(Phases::ConfigureEtcdCa, [config.etcd_hosts.first], ssh: true, parallel: false)
+        apply_phase(Phases::ConfigureEtcdChanges, [config.etcd_hosts.first], ssh: true, parallel: false)
+        apply_phase(Phases::ConfigureEtcd, config.etcd_hosts, ssh: true, parallel: true)
       end
 
-      apply_phase(Phases::ConfigureSecretsEncryption, master_hosts, ssh: true, parallel: false)
-      apply_phase(Phases::ConfigureMaster, master_hosts, ssh: true, parallel: false)
-      apply_phase(Phases::MigrateWorker, config.worker_hosts, ssh: true, parallel: true, master: master_hosts.first)
```

Comment:
(This `master: ...` was already removed as part of #302, but snuck back in during the #293 merge)